market/deal_ingest_snap.go

```diff
@@ -368,14 +409,23 @@ func (p *PieceIngesterSnap) allocateToExisting(ctx context.Context, piece lpiece
 			sec := sec
 			if sec.currentSize+psize <= abi.PaddedPieceSize(p.sectorSize) {
 				if vd.isVerified {
-					sectorLifeTime := sec.latestEndEpoch - sec.earliestStartEpoch
+					si, err := p.api.StateSectorGetInfo(ctx, p.miner, sec.number, types.EmptyTSK)
+					if err != nil {
+						log.Errorw("getting sector info", "error", err, "sector", sec.number, "miner", p.miner)
+						continue
+					}
+
+					sectorLifeTime := si.Expiration - head.Height()
```

Comment:
Isn't this redundant if we check this when we open a new sector?

Reply:
Technically, maybe, but time is moving forward, so it doesn't hurt to double check just in case